docs/semgrep-supply-chain/license-compliance.md

```diff
@@ -0,0 +1,117 @@
+
+## Blocking, commenting, or allowing licenses 
+
+This section provides guides on blocking or allowing packages in CI pipelines based on the license of a package.
+
+### Types of license permissions
+
+Licenses in Semgrep are assigned the following types of permissions:
+
+<dl>
+<dt>Allow</dt>
+<dd>Packages with licenses assigned this type of permission are allowed for use in the codebase.</dd>
+<dt>Comment</dt>
+<dd>Packages with licenses assigned this type of permission are allowed for use in the codebase. A comment is added to the PR or MR. This permission can be useful when you want to remind or warn developers to use certain licenses for internal use only.</dd>
+<dt>Block</dt>
+<dd>Packages with licenses assigned this type of permission are **not** allowed into the codebase. A comment is added to the PR or MR.</dd>
+</dl>
+
+To change the permissions of packages based on the license:
+
+1. From the Supply Chain page, click **Settings** on the header menu. 
+2. Click <i class="fa-solid fa-toggle-large-on"></i> **Notifications and restrictions** if it is not already enabled.
+3. Browse the available licenses within the **License configuration** section. 
+![Screenshot of license configuration section](/img/sc-license-configuration.png#bordered)
+*Figure 3.* Screenshot of Supply Chain > Settings > License configuration section.
+4. Click the permission (Allow, Comment, or Block) you want to set the license to.
+5. Optional: Block entire categories of licenses by clicking on the **Set all to** drop-down box next to the license category.
+
+### License categories
+
+This section describes license categories and licenses identified by Semgrep Supply Chain.
+
+#### Weak copyleft licenses
+
+Software using a package with a weak copyleft license may have to maintain the same license as the package in specific cases. To clarify when this applies, consult your legal department. Developers typically choose these packages based on individual cases, so it's suggested to monitor their usage to ensure license compliance.
+
+* LGPL-3.0
+* LGPL-2.1
+* MPL-2.0
+* EPL-2.0
+* OSL-3.0
+* EUPL-3.0
+
+#### Copyleft licenses
+
+Software using a package with a copyleft license must maintain the same license as the package. To prevent license complications, developers often avoid packages using these licenses.
+
+* GPL-3.0
+* GPL-2.0
+* AGPL-3.0
+* AGPL-2.0
+* CC-BY-SA-4.0
+* APSL
+
+#### Permissive licenses
+
+Software using a package with a permissive license have minimal restrictions on how the software can be used or modified. This makes permissive licenses popular among developers for their flexibility and ease of use without legal concerns.
+
+* MIT
+* Apache-2.0
+* BSD-3
+* BSD-2
+* BSD-3-Clause
+* BSD-2-Clause
+* CC_BY-4.0
+* WTFPL
+* MS-PL
+* Unlicense
+
+#### Uncategorized licenses
+
+Uncategorized licenses are any licenses not included in the previous categories.
```

Comment:
Can we add an additional note saying how some of the licenses in this category could be copyleft/permissive?

https://semgrepinc.slack.com/archives/C0547CY2GJD/p1685994210362589

We don't have any metadata to help us bucket licenses into the three categories above, so we just dumped them all under uncategorized. So maybe say something like "licenses here either don't fit into any of the previous categories, or haven't been categorized yet." You can wordsmith it a bit, but I want to be honest to users that we just haven't categorized a bunch yet lol

Reply:
Added admonition in.

Reply:
added!